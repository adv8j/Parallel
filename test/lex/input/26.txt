func example_function int (int a, int b){
	c = a + b;
	a = 5;
	b = 10;
	return c;
}

func func2 int (int &a, int &b)
{
    c=a*b;
    a=55;
    b=66;
}

struct Person {
    // Data members
        int age;
        float height;
};

//Create an instance of person
Person person;
//Accessing and modifying struct members
person.age = 30;
person.height = 1.75;


Person1=9;
person1=8;
structure=99;


a = 1;
b = 2;
// calling the function
X = example_function(a,b);

y=func2(a,b);
function =5; // variable name function
func = 5; // trying to name a variable func 

dikshant&k=7;

int v=5;
int &k=v;

&t=v; // auto type inference should be there 
&uttam =5 ; // wrong as its not reference I am trying to name a variable starting with &
