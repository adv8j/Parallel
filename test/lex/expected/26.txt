FUNC
IDENTIFIER
INT
LPAREN
INT
IDENTIFIER
COMMA
INT
IDENTIFIER
RPAREN
LBRACE
IDENTIFIER
ASSIGN
IDENTIFIER
PLUS
IDENTIFIER
SEMICOLON
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
RETURN
IDENTIFIER
SEMICOLON
RBRACE
FUNC
IDENTIFIER
INT
LPAREN
INT
REFERENCE
IDENTIFIER
COMMA
INT
REFERENCE
IDENTIFIER
RPAREN
LBRACE
IDENTIFIER
ASSIGN
IDENTIFIER
MUL
IDENTIFIER
SEMICOLON
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
RBRACE
STRUCT
IDENTIFIER
LBRACE
single_line_comment
INT
IDENTIFIER
SEMICOLON
FLOAT
IDENTIFIER
SEMICOLON
RBRACE
SEMICOLON
single_line_comment
IDENTIFIER
IDENTIFIER
SEMICOLON
single_line_comment
IDENTIFIER
DOT
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
IDENTIFIER
DOT
IDENTIFIER
ASSIGN
FLOAT_LITERAL
SEMICOLON
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
single_line_comment
IDENTIFIER
ASSIGN
IDENTIFIER
LPAREN
IDENTIFIER
COMMA
IDENTIFIER
RPAREN
SEMICOLON
IDENTIFIER
ASSIGN
IDENTIFIER
LPAREN
IDENTIFIER
COMMA
IDENTIFIER
RPAREN
SEMICOLON
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
single_line_comment
FUNC
ASSIGN
INT_LITERAL
SEMICOLON
single_line_comment
IDENTIFIER
REFERENCE
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
INT
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
INT
REFERENCE
IDENTIFIER
ASSIGN
IDENTIFIER
SEMICOLON
REFERENCE
IDENTIFIER
ASSIGN
IDENTIFIER
SEMICOLON
single_line_comment
REFERENCE
IDENTIFIER
ASSIGN
INT_LITERAL
SEMICOLON
single_line_comment
